modules/gapi/test/streaming/gapi_gstreamersource_tests.cpp

```diff
@@ -221,36 +273,56 @@ TEST_P(GStreamerSourceTest, GFrameTest)
                                        "appsink"),
                                 Values(cv::Size(1920, 1080)),
                                 Values(10UL)));
 
+
 INSTANTIATE_TEST_CASE_P(FileEmulatingPipeline, GStreamerSourceTest,
                         Combine(Values("videotestsrc pattern=colors num-buffers=10 ! "
                                        "videorate ! videoscale ! "
-                                       "video/x-raw,width=640,height=420,framerate=3/1 ! "
+                                       "video/x-raw,format=NV12,width=640,height=420,framerate=3/1 ! "
+                                       "appsink",
+                                       "videotestsrc pattern=colors num-buffers=10 ! "
+                                       "videorate ! videoscale ! "
+                                       "video/x-raw,format=GRAY8,width=640,height=420,framerate=3/1 ! "
                                        "appsink"),
                                 Values(cv::Size(640, 420)),
                                 Values(10UL)));
 
+
 INSTANTIATE_TEST_CASE_P(MultipleLiveSources, GStreamerSourceTest,
                         Combine(Values("videotestsrc is-live=true pattern=colors num-buffers=10 ! "
-                                       "videoscale ! video/x-raw,width=1280,height=720 ! appsink "
+                                       "videoscale ! video/x-raw,format=NV12,width=1280,height=720 ! appsink "
+                                       "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                                       "fakesink",
+                                       "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                                       "videoscale ! video/x-raw,format=GRAY8,width=1280,height=720 ! appsink "
                                        "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
                                        "fakesink"),
                                 Values(cv::Size(1280, 720)),
                                 Values(10UL)));
 
+
 INSTANTIATE_TEST_CASE_P(MultipleNotLiveSources, GStreamerSourceTest,
                         Combine(Values("videotestsrc pattern=colors num-buffers=10 ! "
-                                       "videoscale ! video/x-raw,width=1280,height=720 ! appsink "
+                                       "videoscale ! video/x-raw,format=NV12,width=1280,height=720 ! appsink "
+                                       "videotestsrc pattern=colors num-buffers=10 ! "
+                                       "fakesink",
+                                       "videotestsrc pattern=colors num-buffers=10 ! "
+                                       "videoscale ! video/x-raw,format=NV12,width=1280,height=720 ! appsink "
                                        "videotestsrc pattern=colors num-buffers=10 ! "
                                        "fakesink"),
```

Comment:
Maybe, for every test, it would be more clear if you firstly define a string with formatting place, and then format it with MediaFormats needed to test, for example
```
std::string camEmulPipeline = "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
                                       "videorate ! videoscale ! "
                                       "video/x-raw,format={},width=1920,height=1080,framerate=3/1 ! "
                                       "appsink";
...
INSTANTIATE_TEST_CASE_P(CameraEmulatingPipeline, ...
                                             Values(std::format(camEmulPipeline, "NV12"),
                                                       std::format(camEmulPipeline, "GRAY8")
...
```

Reply:
std::format

Reply:
OK I see.. my bad

I didn't find a good way to handle it in C++11

Sorry